src/init.c

```diff
@@ -5,18 +5,25 @@
 
 /* .Call calls */
 extern SEXP method_(SEXP, SEXP, SEXP, SEXP);
-extern SEXP method_call_(SEXP, SEXP, SEXP);
+extern SEXP method_call_(SEXP, SEXP, SEXP, SEXP);
+extern SEXP test_call_(SEXP, SEXP, SEXP, SEXP);
 extern SEXP S7_class_(SEXP, SEXP);
 extern SEXP S7_object_(void);
 extern SEXP prop_(SEXP, SEXP);
 extern SEXP prop_set_(SEXP, SEXP, SEXP, SEXP);
 
+#define CALLDEF(name, n)  {#name, (DL_FUNC) &name, n}
+
 static const R_CallMethodDef CallEntries[] = {
-    {"method_", (DL_FUNC) &method_, 4},
-    {"method_call_", (DL_FUNC) &method_call_, 3},
-    {"S7_object_", (DL_FUNC) &S7_object_, 0},
-    {"prop_", (DL_FUNC) &prop_, 2},
-    {"prop_set_", (DL_FUNC) &prop_set_, 4},
+    CALLDEF(method_, 4),
+    CALLDEF(S7_object_, 0),
+    CALLDEF(prop_, 2),
+    CALLDEF(prop_set_, 4),
+    {NULL, NULL, 0}
+};
+
+static const R_ExternalMethodDef ExternalEntries[] = {
+    CALLDEF(method_call_, 2),
```

Comment:
Can you explain why you made this external?

Reply:
`.External2` is the only interface that preserves the visibility of the returned value, `.Call` and the other interfaces all force visibility on (the "2" in "201"). 

<img width="404" alt="image" src="https://github.com/user-attachments/assets/f0376a1f-65a8-44e4-9afa-bbd6ffc954ef">
<img width="643" alt="image" src="https://github.com/user-attachments/assets/d8ca608b-9f49-41f7-a1d3-4802b1b9e5cb">

Reply:
Good to know!